modules/aleph/src/Aleph/AlephClient.php

```diff
@@ -192,11 +196,17 @@ public function changePin(AlephPatron $patron, $new_pin) {
 
     $options['body'] = 'post_xml=' . $xml->asXML();
 
-    $this->requestRest(
+    $response = AlephRequestResponse::createRequestResponseFromXML($this->requestRest(
       'POST',
       'patron/' . $patron->getId() . '/patronInformation/password',
       $options
-    );
+    ));
+
+    if ($response->success()) {
+      return TRUE;
+    }
+
+    throw new AlephPatronInvalidPin();
```

Comment:
Is there any reason why you want to throw an exception if this fails instead of just returning `FALSE` á la

```php
return $response->success();

Reply:
If it fails down to the client, which it shouldn't, I think it's the right thing to do. 
Do you disagree?